pkg/controller/job/job_controller.go

```diff
@@ -1400,6 +1407,17 @@ func (jm *Controller) manageJob(ctx context.Context, job *batch.Job, jobCtx *syn
 		return active, metrics.JobSyncActionPodsDeleted, err
 	}
 
+	var terminating int32 = 0
+	if onlyReplaceFailedPods(jobCtx.job) {
+		// For PodFailurePolicy specified but PodRecreationPolicy disabled
+		// we still need to count terminating pods for replica counts
+		// But we will not allow updates to status.
+		if jobCtx.terminating == nil {
```

Comment:
can you please write a TODO here that this should be refactored (terminating should be always present in the context) once PodRecreationPolicy is GA (enabled by default)?

Reply:
That's generally what we do when features go GA. I don't think it warrants a TODO, since this is already tracked as part of the KEP process

Reply:
I  know it is not necessary and it should be considered when doing a GA. The TODO would just make the next person's job easier and make it less likely to be overlooked, since the KEP is usually not that specific.